lib/devfs/include/devfs/device.h

```diff
@@ -190,22 +219,157 @@ devtoname(struct device *dev)
 	return dev->name;
 }
 
+/*
+ * Opens the device specified by name.
+ * This may be used to check whether a device exists or not.
+ *
+ * @param name
+ * 	The name of the device
+ * @param mode
+ * 	The open mode (should be handled by the device driver if needed)
+ * @param devp
+ * 	Return parameter. The opened device will be reachable in devp after
```

Comment:
This should be marked as an output parameter. See the example here [1]

[1] https://github.com/unikraft/unikraft/blob/staging/lib/vfscore/vfs.h#L277